lib/teleterm/services/connectmycomputer/connectmycomputer.go

```diff
@@ -324,6 +330,119 @@ type Provisioner interface {
 	DeleteToken(ctx context.Context, token string) error
 }
 
+type NodeJoinWait struct {
+	cfg *NodeJoinWaitConfig
+}
+
+func NewNodeJoinWait(cfg *NodeJoinWaitConfig) (*NodeJoinWait, error) {
+	err := cfg.CheckAndSetDefaults()
+	if err != nil {
+		return nil, err
+	}
+
+	return &NodeJoinWait{cfg: cfg}, nil
+}
+
+// Run grabs the host UUID of an agent from disk and then waits for the node with the given name to
+// show up in the cluster.
+//
+// The Electron app calls this method soon after starting the agent process.
+func (n *NodeJoinWait) Run(ctx context.Context, accessAndIdentity AccessAndIdentity, cluster *clusters.Cluster) (clusters.Server, error) {
+	nodeName, err := n.getNodeNameFromHostUUIDFile(ctx, cluster)
+	if err != nil {
+		return clusters.Server{}, err
+	}
+
+	server, err := n.waitForNode(ctx, accessAndIdentity, cluster, nodeName)
+	if err != nil {
+		return clusters.Server{}, trace.Wrap(err)
+	}
+
+	return clusters.Server{
+		URI:    cluster.URI.AppendServer(server.GetName()),
+		Server: server,
+	}, nil
+}
+
+func (n *NodeJoinWait) getNodeNameFromHostUUIDFile(ctx context.Context, cluster *clusters.Cluster) (string, error) {
+	hostUUIDPath := filepath.Join(n.cfg.AgentsDir, cluster.ProfileName, "data", utils.HostUUIDFile)
```

Comment:
Is there some way to create this path that ensures it stays consistent with whatever piece of code creates the file, which presumably needs to construct the same path?

Reply:
I figured that the file was being created somewhere with Go, and so was wondering if there might be some way to marry the code, however I see that that's not the case. Your refactor seems fine nevertheless.